uwblueprint/jump-math
# (1) feat: edit test sessions

## Notion ticket link

[Edit Test Session](https://www.notion.so/uwblueprintexecs/Edit-Test-Session-78d30bb6d7e748dda73a5005c956d984?pvs=4)



## Implementation description
* Add resolver and mutation for `updateTestSession`
* Update queries, types, and components associated with displaying an assessment row to include fields required for editing test session
* Populate data in distribute assessment flow and set page to last page on edit mode
* Add optional `isDisabled` prop to common components (`DatePicker`, `Table`)
* Update inputs ("Assessment Name", "Classroom" and "Start date") in distribute flow to be disabled if user is not editing an **upcoming** test session
   * Also remove pencil icon from Review page
* Populate class info by default when creating test session inside class 



## Steps to test
1. Edit active and upcoming test session from dashboard, assessment page, or classroom page 

https://github.com/uwblueprint/jump-math/assets/69697180/78d9c7bf-6093-4825-8935-9092991a7e0c

2. Create test session from within a class - should choose same class by default

https://github.com/uwblueprint/jump-math/assets/69697180/9ed099bf-1b0f-4e62-a71a-c2fc22145e48


## What should reviewers focus on?
* No breaking changes
* All entrypoints are considered

note: will add backend validation in follow up PR

## Checklist
